Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/StatisticsSettingsController.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.spring.quadriga.aspects.annotations.AccessPolicies;
+import edu.asu.spring.quadriga.aspects.annotations.CheckedElementType;
+import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
+import edu.asu.spring.quadriga.domain.IStatisticsSettings;
+import edu.asu.spring.quadriga.domain.impl.workbench.StatisticsSettingsBean;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.workbench.StatisticsSettingsService;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class StatisticsSettingsController {
+
+    @Autowired
+    private StatisticsSettingsService statsSettingsService;
+
+    /**
+     * This method loads necessary settings option for statistics page if user
+     * have sufficient permission
+     *
+     * @author Ajay Modi
+     * @return modelandview - ModelAndView Object
+     */
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN,
+            RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/{projectid}/statistics", method = RequestMethod.GET)
+    public ModelAndView publicPageStatisticsSettingsForm(
+            @PathVariable("projectid") String projectid)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+
+        ModelAndView model = new ModelAndView("auth/workbench/publicstatistics");
+        model.getModelMap().put("publicpageprojectid", projectid);
+        List<IStatisticsSettings> statisticsSettingsList = statsSettingsService
+                .getStatisticsSettingsList(projectid);
+        model.getModelMap().put("statistics", statisticsSettingsList);
+
+        StatisticsSettingsBean statisticsSettingsBean = new StatisticsSettingsBean();
+        model.getModelMap().put("statisticsSettingsBean",
+                statisticsSettingsBean);
+
+        return model;
+    }
+
+    /**
+     * This method stores necessary settings option for statistics page if user
+     * have sufficient permission
+     *
+     * @author Ajay Modi
+     * @return json
+     */
+
+    @RequestMapping(method = RequestMethod.POST, value = "auth/workbench/{projectid}/submitstatistics")
+    public ModelAndView submitStatisticsSettings(
+            @ModelAttribute("statistics") StatisticsSettingsBean statisticsSettingsBean,
+            BindingResult result, @PathVariable("projectid") String projectid,
+            RedirectAttributes attr) throws QuadrigaStorageException,
+            QuadrigaAccessException {
+        ModelAndView model = new ModelAndView("redirect:/auth/workbench/"
+                + projectid + "/statistics");
+

```

Comment:
just make one call to get the project from the projectid here and add it into the map